pennylane/templates/subroutines/qsvt.py

```diff
@@ -284,9 +284,32 @@ def __init__(self, UA, projectors, id=None):
         total_wires = ua_wires.union(proj_wires)
         super().__init__(wires=total_wires, id=id)
 
+    @property
+    def data(self):
+        r"""Flattened list of operator data in this QSVT operation.
+
+        This ensures that the backend of a ``QuantumScript`` which contains a
+        ``QSVT`` operation can be inferred with respect to the types of the
+        ``QSVT`` block encoding and projector-controlled phase shift data.
+        """
+        return tuple(datum for op in self._operators for datum in op.data)
+
+    @data.setter
+    def data(self, new_data):
+        if new_data:
+            for op in self._operators:
+                if op.num_params > 0:
+                    op.data = new_data[: op.num_params]
+                    new_data = new_data[op.num_params :]
```

Comment:
Is this a new, required part of the operator API? I thought we dropped this in favour of the flatten and unflatten methods @albi3ro?

Reply:
@Jaybsoni My understanding is that this is not strictly required and we only need it here because QSVTs do not have any parameters (only hyper-parameters) which hinders the ability of a QuantumScript to infer the correct backend in the absence of other operators.